packages/common/sftp.js

```diff
@@ -0,0 +1,253 @@
+'use strict';
+
+const { Client } = require('ssh2');
+const { PassThrough, Readable } = require('stream');
+const { lookupMimeType } = require('./util');
+const log = require('./log');
+const {
+  buildS3Uri, getS3Object, getS3ObjectReadStream, promiseS3Upload, s3ObjectExists
+} = require('./aws');
+
+const { KMSProvider: KMS, DefaultProvider } = require('./key-pair-provider');
+
+class Sftp {
+  constructor(config) {
+    // use to indicate an active connection exists
+    this.connected = false;
+    // indicate username and password have been decrypted in options
+    this.decrypted = false;
+    // indicate username and password provided are encrypted
+    this.encrypted = config.encrypted || false;
+    this.username = config.username;
+    this.password = config.password;
+    this.options = {
+      host: config.host,
+      port: config.port || 22,
+      username: config.username,
+      password: config.password,
+      privateKey: config.privateKey,
+      cmKeyId: config.cmKeyId
+    };
+
+    this.client = null;
+    this.sftp = null;
+  }
+
+  async connect() {
+    if (!this.decrypted && this.encrypted) {
+      if (this.password) {
+        this.options.password = await DefaultProvider.decrypt(this.password);
+      }
+
+      if (this.username) {
+        this.options.username = await DefaultProvider.decrypt(this.username);
+      }
+      this.decrypted = true;
+    }
+
+    if (this.options.privateKey) {
+      const bucket = process.env.system_bucket;
+      const stackName = process.env.stackName;
+      // we are assuming that the specified private key is in the S3 crypto directory
+      const keyExists = await s3ObjectExists(
```

Comment:
We need to document how to set this up.

Reply:
It's in docs/ems_reporting.md step 3, is that enough.